src/app/rpmostree-builtin-status.c

```diff
@@ -77,22 +77,32 @@ get_active_txn (RPMOSTreeSysroot *sysroot_proxy)
 }
 
 static void
-print_packages (const char *k, guint max_key_len, const char *const* pkgs)
+print_packages (const char *k, guint max_key_len,
+                const char *const* pkgs,
+                const char *const* omit_pkgs)
 {
   g_autofree char *packages_joined = NULL;
   g_autoptr(GPtrArray) packages_sorted =
     g_ptr_array_new_with_free_func (g_free);
   for (char **iter = (char**) pkgs; iter && *iter; iter++)
     {
-      if (strchr (*iter, ' ') != NULL)
-        g_ptr_array_add (packages_sorted, g_strdup_printf ("'%s'", *iter));
-      else
+      if (omit_pkgs != NULL && g_strv_contains (omit_pkgs, *iter))
+        continue;
+
+      /* don't quote if it just has common pkgname/shell-safe chars */
+      if (g_regex_match_simple ("^[[:alnum:]-._]+$", *iter, 0, 0))
```

Comment:
Since this is in a loop, I think we should take the step of compiling the regexp explicitly.